src/routes/Detail.js

```diff
@@ -1,9 +1,45 @@
 import React from 'react';
+import RepositoryDetailModal from 'components/RepositoryDetailModal';
+import axios from 'axios';
 
 
 export default class Detail extends React.Component {
+
+  state = { repository: null };
+
+  static contextTypes = {
+    router: React.PropTypes.func.isRequired
+  }
+  close = () => {
+    this.context.router.transitionTo('/search', null, { query: this.context.router.getCurrentQuery().query });
+  }
+  getModal = () => (
+    <RepositoryDetailModal repository={this.state.repository} closeModal={this.close.bind(this)} />
+  )
+  fetchDetailedRepository = (owner, repos) => {
+    //TODO: Handle errors and add load spinner in the modal
```

Comment:
very good idea, why not do it now? :)
- handle the `isFetching` and `errors` as it was done in the `SearchResultsPanel`
- show spinner and error message in the modal (pass new props to `RepositoryModal`)